src/states/node.rs

```diff
@@ -1015,27 +1023,28 @@ impl Node {
             return;
         };
 
-        if let Err(error) = self.peer_mgr
-            .verify_candidate(&name, proof, leading_zero_bytes, RESOURCE_PROOF_DIFFICULTY) {
-            debug!("{:?} Failed to verify candidate {}: {:?}",
-                   self,
-                   name,
-                   error);
-            self.candidate_timer_token = None;
-            return;
-        }
-        // For mock-crust tests, just send the `CandidateApproval` immediately.  Otherwise, wait for
-        // the timer to trigger sending the message in an effort to roughly synchronise this.
-        if cfg!(feature = "use-mock-crust") {
-            info!("{:?} Candidate {} passed our challenge.  Sending CandidateApproval.",
-                  self,
-                  name);
-            self.candidate_timer_token = None;
-            let _ = self.send_candidate_approval();
-        } else {
-            info!("{:?} Candidate {} passed our challenge.  Waiting to send CandidateApproval.",
-                  self,
-                  name);
+        match self.peer_mgr.verify_candidate(&name, proof, leading_zero_bytes) {
+            Err(error) => {
+                debug!("{:?} Failed to verify candidate {}: {:?}",
+                       self,
+                       name,
+                       error);
+                self.candidate_timer_token = None;
+            }
+            Ok((target_size, difficulty)) if difficulty == 0 && target_size < 1000 => {
```

Comment:
We might want to set `difficulty` to 0 and have a large `target_size`, or have a small `target_size` but a high difficulty.
I think the right criterion for sending the response immediately is whether the challenge is completely trivial (so it doesn't hinder message accumulation) or not. That's why I'd prefer to have both in there.